### PR TITLE
Allow specifying CPU cores of the VM

### PR DESCRIPTION
**Pull request by rmohr**
Specifying the number is now possible like this:

```yaml
spec:
  domain:
    cpu:
      cores: 3
```
The number of cores gets mapped to a `vcpu` count of `3` with a `static` vcpu mapping. The cpu topology will always have the specified number of cores in combination with a static thread and socket number of 1.

Domain XML example:

```xml
  <vcpu placement='static'>3</vcpu>
  <cpu>
    <topology sockets='1' cores='3' threads='1'/>
  </cpu>
```

The cpu model is not touched. It will be set to the qemu defaults.

## Release Notes

```
It is now possible to specify the amout of cpu cores the vm should have within spec.domain.cpu.
```

### CODE REVIEW

**Comment by rmohr** [comment]
@berrange is this kind of libvrit mapping what you had in mind for the initial stage 1 implementation of cpu cores?

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
@michalskrivanek also your input would be great.

**Comment by rmohr** [comment]
retest this please

**Comment by michalskrivanek** [comment]
fine with me, but worth asking @MarSik for thoughts about properly representing numa topology from the start, as once you have that there is no need for separate "global" sockets/cores. Maybe it makes sense to express simple setups as a single numa node systems.

File: pkg/virt-handler/virtwrap/api/converter.go [hunk 1]
```diff
@@ -325,6 +325,18 @@ func Convert_v1_VirtualMachine_To_api_Domain(vm *v1.VirtualMachine, domain *Doma
...
 	}
 
+	if vm.Spec.Domain.CPU != nil {
+		domain.Spec.CPU.Topology = &CPUTopology{
+			Sockets: 1,
+			Cores:   vm.Spec.Domain.CPU.Cores,
+			Threads: 1,
+		}
+		domain.Spec.VCPU = &VCPU{
+			Placement: "static",
```
Comment (davidvossel) [inline pkg/virt-handler/virtwrap/api/converter.go]: So, with static placement without specifying the cpuset, we're just pinning the qemu process to all visible cores.

This seems fine. In the future we're planning on cpu pinning and numa to be something handled by the container runtime... right?  If that's the case, I think our work's done here for the domain xml part.

Comment (fabiand) [reply pkg/virt-handler/virtwrap/api/converter.go]: Yes, we should work on being able to inherit cpu pinning featrues of the container runtime.

**Comment by fabiand** [comment]
ci test please

**Comment by fabiand** [comment]
retest

**Comment by rmohr** [comment]
retest this please

File: pkg/virt-handler/virtwrap/api/converter.go (hunk 1 above)
Comment (rmohr) [reply pkg/virt-handler/virtwrap/api/converter.go]: Yes, until we have something in k8s which allows pinning, we are done. However once we have that, we will still have to do the pinning for individual cores inside the assigned cpuset.

Comment (fabiand) [reply pkg/virt-handler/virtwrap/api/converter.go]: @nertpinx @ehabkost - Just to put this onto your radar. This will be something interesting to solve.

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Pull request merged**